src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFSurface3DChartData.java

```diff
@@ -0,0 +1,168 @@
+    private CTSurface3DChart chart;
+
+    public XDDFSurface3DChartData(CTSurface3DChart chart, Map<Long, XDDFChartAxis> categories,
+            Map<Long, XDDFValueAxis> values) {
+        this.chart = chart;
+        for (CTSurfaceSer series : chart.getSerList()) {
+            this.series.add(new Series(series, series.getCat(), series.getVal()));
+        }
+        defineAxes(categories, values);
+    }
+
+    private void defineAxes(Map<Long, XDDFChartAxis> categories, Map<Long, XDDFValueAxis> values) {
+        if (chart.sizeOfAxIdArray() == 0) {
+            for (Long id : categories.keySet()) {
+                chart.addNewAxId().setVal(id);
+            }
+            for (Long id : values.keySet()) {
+                chart.addNewAxId().setVal(id);
+            }
+        }
+        defineAxes(chart.getAxIdArray(), categories, values);
+    }
+    
+    public void setSeriesAxisId(XDDFSeriesAxis seriesAxis)
+    {
+            chart.addNewAxId().setVal(seriesAxis.getId());
+    }
+    
+    public CTBoolean getWireframe()
+    {
+        if(chart.isSetWireframe())
+            return chart.getWireframe();
+        else
+            return chart.addNewWireframe();
+    }
+    
+    public void setWireframe(boolean val)
+    {
+        if(chart.isSetWireframe())
+            chart.getWireframe().setVal(val);
+        else
+            chart.addNewWireframe().setVal(val);
+    }
+
+    @Override
+    public void setVaryColors(boolean varyColors) {
+       
+    }
+
+    @Override
+    public XDDFChartData.Series addSeries(XDDFDataSource<?> category,
+            XDDFNumericalDataSource<? extends Number> values) {
+        final int index = this.series.size();
+        final CTSurfaceSer ctSer = this.chart.addNewSer();
+        ctSer.addNewCat();
+        ctSer.addNewVal();
+        ctSer.addNewIdx().setVal(index);
+        ctSer.addNewOrder().setVal(index);
+        final Series added = new Series(ctSer, category, values);
+        this.series.add(added);
+        return added;
+    }
+
+    public class Series extends XDDFChartData.Series {
+        private CTSurfaceSer series;
+
+        protected Series(CTSurfaceSer series, XDDFDataSource<?> category,
+                XDDFNumericalDataSource<? extends Number> values) {
+            super(category, values);
+            this.series = series;
+        }
+
+        protected Series(CTSurfaceSer series, CTAxDataSource category, CTNumDataSource values) {
+            super(XDDFDataSourcesFactory.fromDataSource(category), XDDFDataSourcesFactory.fromDataSource(values));
+            this.series = series;
+        }
+
+        @Override
+        protected CTSerTx getSeriesText() {
+            if (series.isSetTx()) {
+                return series.getTx();
+            } else {
+                return series.addNewTx();
+            }
+        }
+
+        @Override
+        public void setShowLeaderLines(boolean showLeaderLines) {
+            
```

Comment:
add a comment to explain there is no such property as `showLeaderLines` on the underlying `CTSurfaceSer`

Reply:
added comment.